packages/edit-site/src/components/editor/index.js

```diff
@@ -84,44 +86,71 @@ const siteIconVariants = {
 };
 
 export default function EditSiteEditor( { isPostsList = false } ) {
-	const disableMotion = useReducedMotion();
 	const { params } = useLocation();
 	const { canvas = 'view' } = params;
-	const isLoading = useIsSiteEditorLoading();
 	useAdaptEditorToCanvas( canvas );
 	const entity = useResolveEditedEntity();
 	// deprecated sync state with url
 	useSyncDeprecatedEntityIntoState( entity );
 	const { postType, postId, context } = entity;
+	const postWithTemplate = !! context?.postId;
+
+	return (
+		<Editor
+			postType={ postWithTemplate ? context?.postType : postType }
+			postId={ postWithTemplate ? context?.postId : postId }
+			isPostsList={ isPostsList }
+		/>
+	);
+}
+
+export function Editor( { postType, postId, isPostsList = false } ) {
+	const disableMotion = useReducedMotion();
+	const { params } = useLocation();
+	const { canvas = 'view' } = params;
+	const isLoading = useIsSiteEditorLoading();
+	useAdaptEditorToCanvas( canvas );
 	const {
 		supportsGlobalStyles,
 		showIconLabels,
 		editorCanvasView,
 		currentPostIsTrashed,
 		hasSiteIcon,
-	} = useSelect( ( select ) => {
-		const { getEditorCanvasContainerView } = unlock(
-			select( editSiteStore )
-		);
-		const { get } = select( preferencesStore );
-		const { getCurrentTheme, getEntityRecord } = select( coreDataStore );
-		const siteData = getEntityRecord( 'root', '__unstableBase', undefined );
+		templateId,
+	} = useSelect(
+		( select ) => {
+			const { getEditorCanvasContainerView } = unlock(
+				select( editSiteStore )
+			);
+			const { get } = select( preferencesStore );
+			const { getCurrentTheme, getEntityRecord, getTemplateId } = unlock(
+				select( coreDataStore )
+			);
+			const siteData = getEntityRecord(
+				'root',
+				'__unstableBase',
+				undefined
+			);
 
-		return {
-			supportsGlobalStyles: getCurrentTheme()?.is_block_theme,
-			showIconLabels: get( 'core', 'showIconLabels' ),
-			editorCanvasView: getEditorCanvasContainerView(),
-			currentPostIsTrashed:
-				select( editorStore ).getCurrentPostAttribute( 'status' ) ===
-				'trash',
-			hasSiteIcon: !! siteData?.site_icon_url,
-		};
-	}, [] );
-	const postWithTemplate = !! context?.postId;
-	useEditorTitle(
-		postWithTemplate ? context.postType : postType,
-		postWithTemplate ? context.postId : postId
+			return {
+				supportsGlobalStyles: getCurrentTheme()?.is_block_theme,
+				showIconLabels: get( 'core', 'showIconLabels' ),
+				editorCanvasView: getEditorCanvasContainerView(),
+				currentPostIsTrashed:
+					select( editorStore ).getCurrentPostAttribute(
+						'status'
+					) === 'trash',
+				hasSiteIcon: !! siteData?.site_icon_url,
+				templateId: POST_TYPES_WITH_TEMPLATE_SUPPORT.includes(
+					postType
+				)
+					? getTemplateId( postType, postId )
+					: undefined,
+			};
+		},
+		[ postId, postType ]
 	);
+	useEditorTitle( postType, postId );
```

Comment:
Ideally, this hook should be in the route instead, it's the responsibility of the routes to define the "title" of the window, not a random component.